test/com/arpnetworking/metrics/portal/alerts/impl/DatabaseAlertRepositoryTest.java

```diff
@@ -0,0 +1,313 @@
+/**
+ * Copyright 2015 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.alerts.impl;
+
+import com.avaje.ebean.Ebean;
+import com.avaje.ebean.Transaction;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.collect.ImmutableMap;
+import models.ebean.NagiosExtension;
+import models.internal.Alert;
+import models.internal.Context;
+import models.internal.Operator;
+import models.internal.Quantity;
+import models.internal.impl.DefaultAlert;
+import models.internal.impl.DefaultQuantity;
+import org.joda.time.Period;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import play.test.FakeApplication;
+import play.test.Helpers;
+import play.test.WithApplication;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import java.util.UUID;
+import javax.persistence.PersistenceException;
+
+/**
+ * Tests class <code>DatabaseAlertRepository</code>.
+ *
+ * @author Deepika Misra (deepika at groupon dot com)
+ */
+public class DatabaseAlertRepositoryTest extends WithApplication {
+
+    @Override
+    protected FakeApplication provideFakeApplication() {
+        final int port = BASE_PORT + RANDOM.nextInt(1000);
```

Comment:
No. Random ports will be a disaster. When it does eventually collide you won't know it and people will just rerun. My recommendation, and we did it in PCE, was using global increasing port numbers. I would recommend a test helper class that maintained this state and generated the JDBC urls (for example).